hatrac/model/storage/amazons3.py

```diff
@@ -112,24 +116,24 @@ def __init__(self, config):
     def create_from_file(self, name, input, nbytes, content_type=None, content_md5=None, s3_conn=None, s3_bucket=None):
         """Create an entire file-version object from input content, returning version ID."""
         s3_key = boto.s3.key.Key(s3_bucket, name)
-        
         def helper(input, headers, nbytes, md5):
             s3_key.set_contents_from_file(input, headers, replace=True, md5=md5, size=nbytes, rewind=False)
             return s3_key.version_id
-        
         return self._send_content_from_stream(input, nbytes, content_type, content_md5, helper)
 
-    def _send_content_from_stream(input, nbytes, content_type, content_md5, sendfunc):
+    def _send_content_from_stream(self, input, nbytes, content_type, content_md5, sendfunc):
         """Common file-sending logic to talk to S3."""
         headers = {'Content-Length': str(nbytes)}
         if content_type is not None:
             headers['Content-Type'] = content_type
         if content_md5 is not None:
-            md5 = (content_md5, base64.b64encode(binascii.unhexlify(content_md5)))
+            md5 = (content_md5.strip(), base64.b64encode(binascii.unhexlify(content_md5.strip())))
```

Comment:
Hmm, I think with my changes to the internal API, content_md5 will binary hash and what we need here is:

    md5 = (binascii.hexlify(content_md5), base64.b64encode(content_md5))